contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,295 @@
+ - Resource Feasibility Checking
+ - Node Resource Management
+
+Many kubernetes system components currently support these features.  Many more components that support these features are in development.
+The following is not meant to be an exhaustive list, but gives the current set of use cases for these metrics.
+
+Metrics requirements for resource feasibility checking and node resource management, based on kubernetes component needs, are as follows:
+ - Kubelet
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk and Memory
+ - Scheduler (Possibly through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, CPU, and Memory
+  - Pod-level usage metrics for Disk, CPU, and Memory
+  - Container-level usage metrics for Disk, CPU, and Memory
+ - Horizontal-Pod-Autoscaler (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md)))
+  - Node-level capacity and availability metrics for CPU and Memory
+  - Pod-level usage metrics for CPU and Memory
+ - Cluster Federation (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+ - kubectl top (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk, Memory, and CPU
+  - Container-level usage metrics for Disk, CPU, and Memory
+ - Kubernetes Dashboard (Exposed through [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md))
+  - Node-level capacity and availability metrics for Disk, Memory, and CPU
+  - Pod-level usage metrics for Disk, Memory, and CPU
+  - Container-level usage metrics for Disk, CPU, and Memory
+
+
+### Proposed Core Metrics API:
+
+An important difference between the current summary api and the proposed core metrics api is that per-pod stats in the core metrics api contain only usage data, and not capacity-related statistics.  This is more accurate since a pod's resource capacity is really defined by its "requests" and "limits", and it is a better reflection of how components use these metrics.  The kubelet, for example, finds which resources are constrained using node-level capacity and availability data, and then chooses which pods to take action on based on the pod's usage of the constrained resource.  If neccessary, capacity for resources a pod consumes can still be correlated with node-level resources using this format of stats.
```

Comment:
Filesystem capacity for nodeFs, ImageFs, containers and volume filesystems.  It effectively contains capacity for memory since it has availability and usage.